packages/store-sync/src/recs/syncToRecs.ts

```diff
@@ -0,0 +1,136 @@
+import { StoreConfig, storeEventsAbi } from "@latticexyz/store";
+import { Address, Chain, PublicClient, Transport } from "viem";
+import {
+  ComponentValue,
+  Entity,
+  Component as RecsComponent,
+  Schema as RecsSchema,
+  World as RecsWorld,
+  setComponent,
+} from "@latticexyz/recs";
+import { KeySchema, ValueSchema, Table, TableRecord } from "../common";
+import {
+  createBlockStream,
+  isNonPendingBlock,
+  blockRangeToLogs,
+  groupLogsByBlockNumber,
+} from "@latticexyz/block-logs-stream";
+import { filter, map, tap, mergeMap, from, concatMap } from "rxjs";
+import { blockLogsToStorage } from "../blockLogsToStorage";
+import { recsStorage } from "./recsStorage";
+import { hexKeyTupleToEntity } from "./hexKeyTupleToEntity";
+import { debug } from "./debug";
+import { encodeKeyTuple } from "@latticexyz/protocol-parser";
+import { defineInternalComponents } from "./defineInternalComponents";
+import { getTableKey } from "./getTableKey";
+
+type SyncToRecsOptions = {
+  world: RecsWorld;
+  config: StoreConfig;
+  address: Address;
+  // TODO: make this optional and return one if none provided (but will need chain ID at least)
+  publicClient: PublicClient<Transport, Chain>;
+  // TODO: generate these from config and return instead?
+  components: Record<
+    string,
+    RecsComponent<
+      RecsSchema,
+      {
+        contractId: string;
+        keySchema: KeySchema;
+        valueSchema: ValueSchema;
+      }
+    >
+  >;
+  initialState?: {
+    blockNumber: bigint | null;
+    tables: (Table & { records: TableRecord[] })[];
+  };
```

Comment:
I initially tried this with `indexerUrl` and did the indexer fetch here, but that creates a circular dependency between store-sync and store-indexer that I couldn't figure out how to resolve.

(Mostly related to the fact that tRPC expects you to export `typeof appRouter` rather than giving you type helpers to construct the expected type, so defining the app router is coupled to the type you have to export.)